collated/test/A0093896H.md

```diff
@@ -0,0 +1,523 @@
+        
+        t1.addTag(new Tag("school"));
+
+        List<Task> twoTasks = helper.generateTaskList(t1, t2);
+        ToDoList expectedAB = helper.generateToDoList(twoTasks);
+        List<Task> expectedList = helper.generateTaskList(t1);
+        helper.addToModel(model, twoTasks);
+
+        assertCommandBehavior("search tag school",
+                Command.getMessageForTaskListShownSummary(expectedList.size()),
+                expectedAB,
+                expectedList);
+    }
+  
+    
+    @Test
+    public void execute_search_matchesIfBefore() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task t1 = helper.generateTaskWithName("bla bla KEY bla");
+        Task t2 = helper.generateTaskWithName("bla rAnDoM bla bceofeia");
+
+        List<Task> twoTasks = helper.generateTaskList(t1, t2);
+        ToDoList expectedAB = helper.generateToDoList(twoTasks);
+        List<Task> expectedList = helper.generateTaskList(t1, t2);
+        helper.addToModel(model, twoTasks);
+
+        assertCommandBehavior("search before 12/12/2019",
+                Command.getMessageForTaskListShownSummary(expectedList.size()),
+                expectedAB,
+                expectedList);
+    }
+    
+    @Test
+    public void execute_search_matchesIfAfter() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task t1 = helper.generateTaskWithName("bla bla KEY bla");
+        Task t2 = helper.generateTaskWithName("bla rAnDoM bla bceofeia");
+
+        List<Task> twoTasks = helper.generateTaskList(t1, t2);
+        ToDoList expectedAB = helper.generateToDoList(twoTasks);
+        List<Task> expectedList = helper.generateTaskList(t1, t2);
+        helper.addToModel(model, twoTasks);
+
+        assertCommandBehavior("search after 12/12/2013",
+                Command.getMessageForTaskListShownSummary(expectedList.size()),
+                expectedAB,
+                expectedList);
+    }
+    
+    @Test
+    public void execute_search_matchesIfFromTill() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task t1 = helper.generateTaskWithName("bla bla KEY bla");
+        Task t2 = helper.generateTaskWithName("bla rAnDoM bla bceofeia");
+
+        List<Task> twoTasks = helper.generateTaskList(t1, t2);
+        ToDoList expectedAB = helper.generateToDoList(twoTasks);
+        List<Task> expectedList = helper.generateTaskList(t1, t2);
+        helper.addToModel(model, twoTasks);
+
+        assertCommandBehavior("search from 12/12/2013 to 12/12/2019",
+                Command.getMessageForTaskListShownSummary(expectedList.size()),
+                expectedAB,
+                expectedList);
+    }
+    
+    @Test
+    public void execute_search_matchesIfDone() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task t1 = helper.generateTaskWithName("bla bla KEY bla");
```

Comment:
these are definitely duplicated code; use `@Before` and `@After` as necessary and move them to a common setup.